crates/sui-types/src/crypto.rs

```diff
@@ -645,14 +801,39 @@ impl<S: AggregateAuthenticator> VerificationObligation<S> {
         let mut message = Vec::new();
         message_value.write(&mut message);
 
-        self.signatures.push(S::default());
+        self.signatures.push(AggregateAuthoritySignature::default());
         self.public_keys.push(Vec::new());
         self.messages.push(message);
         self.messages.len() - 1
     }
 
+    // Attempts to add signature and public key to the obligation. If this fails, ensure to call `verify` manually.
```

Comment:
I didn't think that it makes too much sense to try batch user signatures for now - this is the cleanest workaround I've come up with so far that shouldn't affect performance in this PR, as it all should be abstractions, but happy to take ideas on other ways to do this.